tests/zfs-tests/tests/functional/fault/auto_online_001_pos.ksh

```diff
@@ -83,53 +84,57 @@ if is_loop_device $DISK1; then
 	log_must $ZPOOL create -f $TESTPOOL raidz1 $SDDEVICE_ID $DISK1 \
 	    $DISK2 $DISK3
 elif ( is_real_device $DISK1 || is_mpath_device $DISK1 ); then
+	#else use the persistent names for sd devices
 	log_must $ZPOOL create -f $TESTPOOL raidz1 ${devs_id[0]} \
 	    ${devs_id[1]} ${devs_id[2]}
 else
 	log_fail "Disks are not supported for this test"
 fi
 
-#add some data to the pool
+#Add some data to the pool
 log_must $MKFILE $FSIZE /$TESTPOOL/data
 
-#pool guid import
-typeset guid=$(get_config $TESTPOOL pool_guid)
-if (( RANDOM % 2 == 0 )) ; then
-	target=$guid
-fi
-
 for offline_disk in $autoonline_disks
 do
 	log_must $ZPOOL export -F $TESTPOOL
 
-	host=$($LS /sys/block/$offline_disk/device/scsi_device | $NAWK -F : '{ print $1}')
-	#offline disk
+	host=$($LS /sys/block/$offline_disk/device/scsi_device \
+	    | $NAWK -F : '{ print $1}')
+
+	#Offline disk
 	on_off_disk $offline_disk "offline"
 
-	#reimport pool with drive missing
-	log_must $ZPOOL import $target
+	#Reimport pool with drive missing
+	log_must $ZPOOL import $TESTPOOL
 	check_state $TESTPOOL "" "degraded"
 	if (($? != 0)); then
 		log_fail "$TESTPOOL is not degraded"
 	fi
 
-	#online disk
+	#Online disk
 	on_off_disk $offline_disk "online" $host
-	
+
+	#Clear zpool events
+	$ZPOOL events -c $TESTPOOL
```

Comment:
This should be done before the disk is brought back online.  If it's done after then it's possible that ZED will have already onlined the device and the resilver could have finished.  In which case you'll throw out the message and timeout below.  This might explain the testing failures.